### ISSUE TITLE
ModuleNotFoundError: No module named 'causalml.inference.tree.causaltree'

### ISSUE THREAD

**Issue by ngdelamo** [uber/causalml#148]
Hi!

I'm trying to run this example notebook:
https://github.com/uber/causalml/blob/master/examples/meta_learners_with_synthetic_data.ipynb

However, when I run the import statements in the 3rd cell, I get the following error:

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-3-e0de716ab558> in <module>
     13 from causalml.match import NearestNeighborMatch, MatchOptimizer, create_table_one
     14 from causalml.propensity import ElasticNetPropensityModel
---> 15 from causalml.dataset import *
     16 from causalml.metrics import *
     17 

~\causalml\src\causalml\causalml\dataset\__init__.py in <module>
      7 from .classification import make_uplift_classification
      8 
----> 9 from .synthetic import get_synthetic_preds, get_synthetic_preds_holdout
     10 from .synthetic import get_synthetic_summary, get_synthetic_summary_holdout
     11 from .synthetic import scatter_plot_summary, scatter_plot_summary_holdout

~\causalml\src\causalml\causalml\dataset\synthetic.py in <module>
     14 
     15 from causalml.inference.meta import BaseXRegressor, BaseRRegressor, BaseSRegressor, BaseTRegressor
---> 16 from causalml.inference.tree import CausalTreeRegressor
     17 from causalml.propensity import ElasticNetPropensityModel
     18 from causalml.metrics import plot_gain, get_cumgain

~\causalml\src\causalml\causalml\inference\tree\__init__.py in <module>
      1 from .models import UpliftTreeClassifier, DecisionTree
      2 from .models import UpliftRandomForestClassifier
----> 3 from  .causaltree import CausalMSE, CausalTreeRegressor
      4 from .plot import uplift_tree_string, uplift_tree_plot
      5 from .utils import cat_group, cat_transform, cv_fold_index, cat_continuous, kpi_transform

ModuleNotFoundError: No module named 'causalml.inference.tree.causaltree'

```

I installed causalml via `pip`, version `0.7.0`.

Thanks in advance!

**Comment by jeongyoonlee** [comment]
Hi @ngdelamo, could you share more info such as OS, Python version, environment (system vs. virtualenv vs. conda), etc.?

FYI, I just tried the pip installation on new conda environments with Python 3.6 and 3.7, and it went through on both environments.

**Comment by ngdelamo** [comment]
Hi @jeongyoonlee, thanks for the quick answer.

I'm currently using Windows 10, and the setup is a virtualenv created with `conda` on python 3.7, but the rest of the packages were installed using `pip`. 

Although, now that I mention it, I remember that when I installed `causalml` through pip, it showed an error saying that I had to install _"Visual Studio buildtools"_, and stopped. I installed the said buildtools (version `16.4.5`), and after that, I re-installed `causalml `through pip, which finished smoothly without errors.

I see that `causalml.inference.tree.causaltree` is a Cython module, maybe these _VS buildtools_ failed to build the module silently?

If it helps you, I could uninstall these VS buildtools, create a new virtualenv, and try again to install `causalml`, to see if I can get a screenshots of the process...

**Comment by ngdelamo** [comment]
Hi @jeongyoonlee.

I have finally managed to make it work. After digging a bit about Cython (I've never used it before) the problem seemed to be that the Cython module wasn't being compiled upon installation. In order to solve it, I cloned the `causalml` Github's repo and build the Cython modules from source via:

```
python setup.py build_ext --inplace
```

After that, the notebooks work properly. Maybe it's an installation problem on Windows platforms?

**Comment by jeongyoonlee** [comment]
Thanks for sharing the details, @ngdelamo. According to [this article](https://www.scivision.dev/python-windows-visual-c-14-required/), it seems that the VS buildtools is required to install a Python package that uses Cython on Windows.

We will update the installation instruction with this information.

Thanks!

**Comment by ngdelamo** [comment]
Thanks for investigating this issue, @jeongyoonlee !

Just for clarification: I indeed saw a message during installation that stated that "VS build tools was required". The problem is that I installed them and the Cython modules were still not compiled. I had to clone the repository and compile the Cython modules myself from the source code using `python setup.py build_ext --inplace` in order to make the notebooks work.

Please, let me know if you want me to do any testing with my environment.

**Comment by jeongyoonlee** [comment]
@ngdelamo, this issue has been fixed with #161. It was because the notebook imported `causalml` from the source code directory even when `causalml` has been installed in the system. However, importing from the source code directory requires compiled `Cython` objects - as you noted, with `python setup.py build_ext --inplace`. 

This fix made the notebook import `causalml` installed in the system instead.

Thanks again for the report.
